### PR TITLE
Sorting v2

### PR DESCRIPTION
**Pull request by mayank23**
converted the sort feature into an actual rule. increases configurability

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -291,6 +307,19 @@ Check out the [Examples](examples/) folder to see different use cases.
 
   ![](assets/identical-keys-error-screenshot-2018-04-30.png)
 
+### i18n-json/sorted-keys
+
+- automatic ascending sort of all keys in the translation file.
+- default severity: error | 2
```
Comment (gergelyke) [inline README.md]: just to keep it consistent, should we use `.` here too? or remove from the rest of the list items too?

File: src/sorted-keys.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  // default traversal is asc.
+  let traversalOrder = keyTraversals.asc;
+
+  if (order.toLowerCase() === 'desc') {
+    traversalOrder = keyTraversals.desc;
+  }
+
+
```
Comment (gergelyke) [inline src/sorted-keys.js]: do we need these empty lines?

File: src/util/key-traversals.test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const keyTraverals = require('./key-traversals');
```
Comment (pnevares) [inline src/util/key-traversals.test.js]: Typo here and same in line 5

File: README.md (hunk 1 above)
Comment (mayank23) [reply README.md]: ah, yup, thanks!

File: src/sorted-keys.js (hunk 2 above)
Comment (mayank23) [reply src/sorted-keys.js]: nope, will remove 😄

File: src/util/key-traversals.test.js (hunk 3 above)
Comment (mayank23) [reply src/util/key-traversals.test.js]: thanks! 😄

**Pull request merged**